blas/impl/KokkosBlas1_axpby_unification_attempt_traits.hpp

```diff
@@ -0,0 +1,955 @@
+
+ public:
+  using InternalTypeA_managed = std::conditional_t<
+      Kokkos::is_view_v<InternalTypeA>,
+      Kokkos::View<InternalScalarTypeA*, InternalLayoutA_managed,
+                   typename XMV::device_type>,
+      void>;
+
+  // ********************************************************************
+  // Declare 'InternalTypeB' if 'InternalTypeA_tmp' is a view then
+  // make sure 'InternalTypeB' is a view as well
+  // ********************************************************************
+  using InternalTypeB = std::conditional_t<
+      Kokkos::is_view_v<InternalTypeA_tmp> &&
+          !Kokkos::is_view_v<InternalTypeB_tmp>,
+      Kokkos::View<const InternalScalarTypeB*, InternalLayoutB,
+                   typename YMV::device_type,
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged>>,
+      InternalTypeB_tmp>;
+
+  // ********************************************************************
+  // Declare 'InternalTypeB_managed' with the same scalar type in
+  // 'InternalTypeB'
+  // ********************************************************************
+ private:
+  using InternalLayoutB_managed = InternalLayoutB;
+
+ public:
+  using InternalTypeB_managed = std::conditional_t<
+      Kokkos::is_view_v<InternalTypeB>,
+      Kokkos::View<InternalScalarTypeB*, InternalLayoutB_managed,
+                   typename YMV::device_type>,
+      void>;
+
+  // ********************************************************************
+  // Auxiliary Boolean results on internal types
+  // ********************************************************************
+ private:
+  static constexpr bool internalTypeA_is_scalar =
+      !Kokkos::is_view_v<InternalTypeA>;
+  static constexpr bool internalTypeA_is_r1d = Tr1d_val<InternalTypeA>();
+
+  static constexpr bool internalTypeB_is_scalar =
+      !Kokkos::is_view_v<InternalTypeB>;
+  static constexpr bool internalTypeB_is_r1d = Tr1d_val<InternalTypeB>();
+
+ public:
+  static constexpr bool internalTypesAB_bothScalars =
+      (internalTypeA_is_scalar && internalTypeB_is_scalar);
+  static constexpr bool internalTypesAB_bothViews =
+      (internalTypeA_is_r1d && internalTypeB_is_r1d);
+
+  // ********************************************************************
+  // Routine to perform checks (both compile time and run time)
+  // ********************************************************************
+  static void performChecks(const AV& a, const XMV& X, const BV& b,
+                            const YMV& Y) {
+    // ******************************************************************
+    // Check 1/6: General checks
+    // ******************************************************************
+    static_assert(
+        Kokkos::is_execution_space_v<tExecSpace>,
+        "KokkosBlas::Impl::AxpbyUnificationAttemptTraits::performChecks()"
+        ": tExecSpace must be a valid Kokkos execution space.");
+
+    static_assert(
+        (xyRank1Case && !xyRank2Case) || (!xyRank1Case && xyRank2Case),
```

Comment:
Why not just assert the following: `static_asser(!xyRank1Case && !xyRank2Case)` which basically means we have a mismatch of ranks between x and y

Reply:
Thanks, Luc. I kept the way it is because the way you suggested we could have the case [ neither rank1 and neither rank2 ] accepted. I think the code is understandable and "clean" the way it is now.